go-packages/sim2/car.go

```diff
@@ -1,163 +1,160 @@
 package sim2
 
 import (
-  //"fmt"
-  "math"
+  "log"
+  "fmt"
 )
 
 // car - Describes routine hooks and logic for Cars within a World simulation
 
+// The distance the car should move every drive call
+const MovementPerDrive = 1.0
+
 // Car - struct for all info needed to manage a Car within a World simulation.
 type Car struct {
   // TODO determine if Car needs any additional/public members
+  id           uint
+  path         Path
+  world        CarWorldInterface
+  syncChan     chan bool
+  sendChan     *chan CarInfo
+  ethApi       EthApiInterface
+  requestState RequestState
+}
+
+type Path struct {
+  pickUp       Location
+  dropOff      Location
+  currentPos   Coords
+  currentDir   Coords
+  currentEdge  Edge
+  currentState PathState
+  routeEdges   []Edge
+  riderAddress string
+}
 
-  id uint
-  dstNode uint
-  path []uint
-  world *World
-  syncChan chan bool
-  sendChan *chan CarInfo
+type Location struct {
+  intersect Coords
+  edge Edge
 }
 
+type PathState int
+const (
+  DrivingAtRandom PathState = 0
+  ToPickUp        PathState = 1
+  ToDropOff       PathState = 2
```

Comment:
I think there's a fourth state of the vehicle that's missing currently: `Stopped`. It would primarily be used at the pickup/dropoff points and whenever the car needs to prevent collision or obey road rules. Adding this state allows us to trigger off of external events to put the car into an active state again (i.e. client user interface interactions, emergency stops, timer + clear road for stop signs, etc).